github/main.js

```diff
@@ -1,552 +1,898 @@
-// Copyright 2017 Google LLC.
-//
-// Licensed under the Apache License, Version 2.0 (the "License"); you may not
-// use this file except in compliance with the License. You may obtain a copy of
-// the License at
-//
-//     https://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an 'AS IS' BASIS, WITHOUT
-// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
-// License for the specific language governing permissions and limitations under
-// the License.
+'use strict';
 
 /**
- * @fileoverview Community Connector for GitHub. Retrieves historical star data
- * for a GitHub repository.
+ * @license
+ * Copyright 2018 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0*
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
  */
 
-/** define namespace */
-var connector = connector || {};
-
-/** @const */
-connector.STARRED_AT = 'starred_at';
-
-/** @const */
-connector.LINK_KEY = 'Link';
-
-/** @const */
-connector.OAUTH_CONST = 'OAUTH2';
-
-/** @const */
-connector.OAUTH_CLIENT_ID = 'OAUTH_CLIENT_ID';
-
-/** @const */
-connector.OAUTH_CLIENT_SECRET = 'OAUTH_CLIENT_SECRET';
+/**
+ * HELPERS: THESE FUNCTIONS ARE GENERAL PURPOSE HELPERS FOR THE REST
+ * OF THE COMMUNITY CONNECTORS
+ */
 
-/** @const */
-connector.logEnabled = false;
+//TODO(nathanwest): Roll this stuff into a separate library
 
-/** @const */
-connector.defaultOrganization = 'google';
+/**
+ * Similar to Array.map, but for an object. Maps the value of each property to
+ * an array, in sorted order by property name.
+ * @param  {!Object<string, T>} object The object to map over
+ * @param  {!function(string, T, Object<string, T>): U} func The mapping
+ *   function. Called once with (property value, property name, object) for
+ *   each of the object properties, as returned by Object.keys
+ * @return {!Array<U>} The mapped array with the return values of func.
+ * @template T, U
+ * @example
+ * var obj = {c: 3, b: 2, a: 1}
+ * var result = mapObjectSorted(obj, function(value) { return value + 1})
+ * // result == [2,3,4]
+ */
+function mapObjectSorted(object, func) {
```

Comment:
This is only used in one place, could you just inline it?

Reply:
Oh, is it? I thought I used it more than once. Sure.